doc/cabal-commands.rst

```diff
@@ -496,38 +496,23 @@ flag.
 cabal freeze
 ^^^^^^^^^^^^
 
-If a package is built in several different environments, such as a
-development environment, a staging environment and a production
-environment, it may be necessary or desirable to ensure that the same
-dependency versions are selected in each environment. This can be done
-with the ``freeze`` command:
-
-``cabal freeze`` writes out a **freeze file** which records all of
-the versions and flags that are picked by the solver under the
-current index and flags.  Default name of this file is
-``cabal.project.freeze`` but in combination with a
-``--project-file=my.project`` flag (see :ref:`project-file
-<cmdoption-project-file>`)
-the name will be ``my.project.freeze``.
-A freeze file has the same syntax as ``cabal.project`` and looks
-something like this:
-
-.. highlight:: cabal
+.. code-block:: console
 
-::
+    $ cabal freeze
 
-    constraints: HTTP ==4000.3.3,
-                 HTTP +warp-tests -warn-as-error -network23 +network-uri -mtl1 -conduit10,
-                 QuickCheck ==2.9.1,
-                 QuickCheck +templatehaskell,
-                 -- etc...
+generates ``cabal.project.freeze`` file, which describes the exact dependency
+tree as it was resolved at that moment by Cabal.  This means it captures an
+exact version of every dependency, including dependencies of dependencies,
+recursively all the way.
 
+Since ``cabal`` reads ``cabal.project.freeze`` when present, and takes into
+consideration the version constraints in it, this means that by producing
+``cabal.project.freeze`` you are guaranteed that every future ``cabal`` call
+will use the exact same set of dependencies, regardless of any updates (even
+patches) that might get published for these dependencies in the meantime.
+Therefore, we have effectively "frozen" the dependencies in place.
```

Comment:
Please also mention that while `cabal.project.freeze` restricts specified dependency versions, it does NOT prevent from including future dependencies unless `reject-unconstrained-dependencies=all` is specified